# Microbiome Analysis

## Description

These sessions will cover the use of a variety of software tools needed for the analysis of microbiome data, from the handling of the Illumina sequencing data, to the processing of 16S rRNA amplicon data. During these sessions, the students will be able to:

* Evaluate the quality of an Illumina sequencing run, including data filtering;
* Carry out assemblies of 16S rRNA amplicons;
* Learn the basics on the R programming language and environment; and 
* State how to manipulate microbiome data including count tables and sample metadata. 

#### Prerequisites

Participants must have already taken a basic course in Bioinformatics.

#### Venue and Dates

[IBIMA](https://ibima.med.uni-rostock.de) Computerraum ([Ernst-Heydemann-Str. 8](https://goo.gl/maps/JGDWhPDLHxG2), 3. Etage, Nr. 3016) <br>
17.10 and 24.10.2018, 10:00 - 11:30 hrs.

## Materials and Data

#### Software

To carry on with the analyses, the students the following programs already installed before the Praktikum sessions:

* RStudio: https://www.rstudio.com/products/rstudio/download/
* Pandaseq: https://github.com/neufeld/pandaseq/wiki/Installation

Please notice that pandaseq requires certain libraries in Linux and Mac OSX, as indicated on the link above.

#### Online tools

* European [Galaxy mirror](https://usegalaxy.eu)
* [RDP Aligner](https://pyro.cme.msu.edu/aligner/form.spr)
* [RDP Complete Clustering](https://pyro.cme.msu.edu/cluster/form.spr) tool

#### Experimental Data

* Slides and data for the course: [link](http://tiny.cc/UMRMicrobiome2018) (this will expire shortly after the Praktikum sessions).

Alternatively, the course can be also followed using the [MiSeq SOP](http://www.mothur.org/w/images/d/d6/MiSeqSOPData.zip) FASTQ files.

## References

* Afgan E et al. (2018). The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2018 update. Nucleic Acids Research, 46(W1), W537-W544. https://doi.org/10.1093/nar/gky379
* Cole JR et al. (2014). Ribosomal Database Project: data and tools for high throughput rRNA analysis. Nucleic Acids Research, 42(D1), D633–D642. https://doi.org/10.1093/nar/gkt1244
* Knight R et al. (2018). Best practices for analysing microbiomes. Nature Reviews Microbiology, 1. https://doi.org/10.1038/s41579-018-0029-9
* Goodrich JKK et al. (2014). Conducting a Microbiome Study. Cell, 158(2), 250–262. https://doi.org/10.1016/j.cell.2014.06.037
* Masella AP et al. (2012). PANDAseq: Paired-end assembler for illumina sequences. BMC Bioinformatics, 13(1), 31. https://doi.org/10.1186/1471-2105-13-31
* McDonald D et al. (2012). The Biological Observation Matrix (BIOM) format or: how I learned to stop worrying and love the ome-ome. GigaScience, 1(1), 7. https://doi.org/10.1186/2047-217X-1-7
* McMurdie PJ and Holmes S. (2013). Phyloseq: An R Package for Reproducible Interactive Analysis and Graphics of Microbiome Census Data. PLoS ONE, 8(4), e61217. https://doi.org/10.1371/journal.pone.0061217
* Callahan BJ et al. (2016). Bioconductor workflow for microbiome data analysis: from raw reads to community analyses. F1000Research, 5, 1492. https://doi.org/10.12688/f1000research.8986.1

## Contact

Dr. rer. nat. Israel Barrantes <br>
Institute for Biostatistics and Informatics in Medicine and Ageing Research, Office 3017<br>
Rostock University Medical Center<br>
Ernst-Heydemann-Str. 8<br>
18057 Rostock, Germany<br>

Email: israel.barrantes[at]uni-rostock.de


---

Last update: 2018/10/17
