# Microbiome Analysis

## Description

These sessions will cover the use of a variety of software tools needed for the analysis of microbiome data, from the handling of the Illumina sequencing data, to the processing of 16S rRNA amplicon data. During these sessions, the students will be able to:

* Evaluate the quality of an Illumina sequencing run, including data filtering;
* Carry out assemblies of 16S rRNA amplicons;
* Learn the basics on the R programming language and environment; and 
* State how to manipulate microbiome data including count tables and sample metadata. 

#### Prerequisites

Participants must have already taken a basic course in Bioinformatics.

#### Venue and Dates

IBIMA Computerraum (Ernst-Heydemann-Str. 8, 3. Etage, Nr. 3016)
17.10 and 24.10.2018, 10:00 - 11:30 hrs.

## Materials and Data

#### Software

To carry on with the analyses, the students the following programs already installed before the Praktikum sessions:

* RStudio: https://www.rstudio.com/products/rstudio/download/
* Pandaseq: https://github.com/neufeld/pandaseq/wiki/Installation

#### Online tools

* European [Galaxy mirror](https://usegalaxy.eu)
* [RDP Aligner](https://pyro.cme.msu.edu/aligner/form.spr)
* [RDP Complete Clustering](https://pyro.cme.msu.edu/cluster/form.spr) tool

#### Experimental Data

* Slides and data for the course: [link](http://tiny.cc/UMRMicrobiome2018) (this will expire shortly after the Praktikum sessions).

Alternatively, the course can be also followed using the [MiSeq SOP](http://www.mothur.org/w/images/d/d6/MiSeqSOPData.zip) FASTQ files.

## Contact

Dr. rer. nat. Israel Barrantes <br>
Institute for Biostatistics and Informatics in Medicine and Ageing Research, Office 3017<br>
Rostock University Medical Center<br>
Ernst-Heydemann-Str. 8<br>
18057 Rostock, Germany<br>

Email: israel.barrantes[at]uni-rostock.de


---

Last update: 2018/10/17
